### ISSUE TITLE
🐛 BUG: pages_build_output_dir resolving to root

### ISSUE THREAD

**Issue by WalshyDev** [cloudflare/workers-sdk#5983]
### Which Cloudflare product(s) does this pertain to?

Pages

### What version(s) of the tool(s) are you using?

3.50.0

### What version of Node are you using?

20

### What operating system and version are you using?

Pages CI

### Observed behavior
If you have a config like:
```
pages_build_output_dir = "/dist"
```

It will resolve to the root of Pages CI (and thus throw an error). It will also fail to resolve in wrangler dev and cause the assets to not be served.

Example bad build: 9feb8399-ddc5-473b-bf90-0776a8f7ca62

### Expected behavior
It should resolve as if it was `./` (as in, within the defined root). This is the behaviour today on Pages and we should continue it through wrangler.toml

### Steps to reproduce

Set:
```
pages_build_output_dir = "/dist"
```

and try to run a build through CI or run `wrangler pages dev`

**Comment by petebacondarwin** [comment]
> It should resolve as if it was ./ (as in, within the defined root). This is the behaviour today on Pages and we should continue it through wrangler.toml

I don't think this is a good idea. The other paths in wrangler.toml are relative to the wrangler.toml path (or possibly the current working directory).
Changing the path resolution just for this field is inconsistent with the other paths. But moreover, it would prevent users from providing an absolute path in this field. (I appreciate that using an absolute path here would be bad practice in the context of CI since the user doesn't control the containing paths to the project.)

**Comment by irvinebroque** [comment]
Better error message from our API, and making sure that API error gets surfaced in Wrangler CLI is what is needed here
